tests/devices/i04/test_transfocator.py

```diff
@@ -1,66 +1,75 @@
-from unittest.mock import MagicMock, patch
+import asyncio
+from unittest.mock import AsyncMock, patch
 
 import pytest
-from ophyd.sim import make_fake_device
-from ophyd.status import SubscriptionStatus
+from ophyd_async.core import (
+    DeviceCollector,
+    set_mock_value,
+    wait_for_value,
+)
 
 from dodal.devices.i04.transfocator import Transfocator
 
 
 @pytest.fixture
-def fake_transfocator():
-    FakeTransfocator = make_fake_device(Transfocator)
-    transfocator: Transfocator = FakeTransfocator(name="test_transfocator")
-    yield transfocator
+async def fake_transfocator() -> Transfocator:
+    async with DeviceCollector(mock=True):
+        transfocator = Transfocator(prefix="", name="transfocator")
+    return transfocator
 
 
 def given_predicted_lenses_is_half_of_beamsize(transfocator: Transfocator):
     def lens_number_is_half_beamsize(value, *args, **kwargs):
-        transfocator.predicted_vertical_num_lenses.sim_put(int(value / 2))  # type: ignore
+        set_mock_value(transfocator.predicted_vertical_num_lenses, int(value / 2))
 
-    transfocator.beamsize_set_microns.subscribe(lens_number_is_half_beamsize)
+    transfocator.beamsize_set_microns.subscribe_value(lens_number_is_half_beamsize)
 
 
-def test_given_beamsize_already_set_then_when_transfocator_set_then_returns_immediately(
+async def set_beamsize_to_same_value_as_mock_signal(
+    transfocator: Transfocator, value: float
+):
+    transfocator.set(value)
+    set_mock_value(transfocator.beamsize_set_microns, value)
+
+
+async def test_given_beamsize_already_set_then_when_transfocator_set_then_returns_immediately(
     fake_transfocator: Transfocator,
 ):
-    fake_transfocator.beamsize_set_microns.sim_put(100)  # type: ignore
-    set_status = fake_transfocator.set(100)
-    set_status.wait(0.01)
-    assert set_status.done and set_status.success
+    await asyncio.wait_for(
```

Comment:
Should: Assert that none of the other logic, (eg `set_based_on_predicition`) was ever ran. I think that would have caught the above function having sets in the wrong order